filter/series_by_tag.go

```diff
@@ -130,6 +130,10 @@ func (index *SeriesByTagPatternIndex) MatchPatterns(name string, labels map[stri
 		}
 	}
 
+	if len(index.filtersByTag) != len(matchedPatterns) {
```

Comment:
Could you explain why we need this?

Reply:
In my testing, sending two metrics:
```
metricOne.test;region=us-east-2;az=a;service=foo
metricTwo.test;region=us-east-2;az=a;service=foo
```

Having the following trigger target:
```
seriesByTag("name=metricOne.test", 'region=us-east-2', "az=a", "service=foo")
```

Make both metrics pass the filter and arrive checker, when only the first should.

My conclusion is that 
```
func (index *SeriesByTagPatternIndex) MatchPatterns
```
returns metrics that matched any of the given tags. My understanding is that it should only return metrics that match all given tags, because of that a check that compare the number off filtertags and matched patterns.

Reply:
After additional testing the main thing here (before this pr) is that even when metric doesn't match name it keep going and if there's return specs that match any tag.
Other way to fix is to simply return if name doesn't match.

Besides that after advancing with test I realized that tag with empty value (that should match series that does not have the tag) also does not work, series keep matching...

Reply:
Yep, I will fix it.

Reply:
Actually, it seems your fix is not correct in general, but I will try to rework it, so many thanks for reporting the issue.

Reply:
BTW Here is a test https://github.com/moira-alert/moira/pull/294/commits/469a511fd8ac48a70b9602c8eecbb0edbc697b20

Reply:
agreed, looking the code for more time I understand kkkk

Reply:
Here is a rework of the `SeriesByTagPatternIndex` https://github.com/moira-alert/moira/pull/294/commits/8e60157d3c2ed2d6640684a326514ecfd6558ee4. It is simple enough but I hope it covers most of the cases.

Reply:
Also we added single quoted tag spec https://github.com/moira-alert/moira/pull/294/commits/6af3cbd738dd3278987cbddc8b416623f360ef70